app/src/main/java/com/example/instagram/MainActivity.java

```diff
@@ -35,178 +39,37 @@
-    private EditText etDescription;
-    private Button btnCaptureImage;
-    private ImageView ivPostImage;
-    private Button btnSubmit;
-    public String photoFileName = "photo.jpg";
-    private File photoFile;
+
+    final FragmentManager fragmentManager = getSupportFragmentManager();
     private BottomNavigationView bottomNavigationView;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        etDescription = findViewById(R.id.etDescription);
-        btnCaptureImage = findViewById(R.id.btnCaptureImage);
-        ivPostImage = findViewById(R.id.ivPostImage);
-        btnSubmit = findViewById(R.id.btnSubmit);
-        btnSubmit.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                String description = etDescription.getText().toString();
-                if (description.isEmpty()) {
-                    Toast.makeText(MainActivity.this, "Description cannot be empty", Toast.LENGTH_SHORT).show();
-                    return;
-                }
-                if (photoFile == null || ivPostImage.getDrawable() == null) {
-                    Toast.makeText(MainActivity.this, "There is no image!", Toast.LENGTH_SHORT).show();
-                    return;
-                }
-                ParseUser currentUser = ParseUser.getCurrentUser();
-                savePost(description, currentUser, photoFile);
-
-                // make submit button go to feed
-                Intent i = new Intent(MainActivity.this, FeedActivity.class);
-                startActivity(i);
-            }
-        });
-
-        btnLogout = findViewById(R.id.btnLogout);
-        btnLogout.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Log.i(TAG, "onClick logout button");
-                ParseUser.logOutInBackground();
-                ParseUser currentUser = ParseUser.getCurrentUser(); // this will now be null
-                logoutUser();
-            }
-        });
-
-        btnCaptureImage.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View v) {
-                launchCamera();
-            }
-        });
-
         bottomNavigationView = findViewById(R.id.bottomNavigation);
         bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
             @Override
             public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                 Fragment fragment;
                 switch (item.getItemId()) {
                     case R.id.action_home:
-                        Toast.makeText(MainActivity.this, "Home!", Toast.LENGTH_SHORT).show();
-                        return true;
+                        fragment = new PostsFragment();
+                        break;
```

Comment:
This looks a bit funky for a switch statement that is intended to return a boolean; that is, having `break` after each case and then `default:return true;`. I would expect default case to return false, as we're reaching that block if we haven't encountered a view ID to handle, and the intention is to [return true to display the item as the selected item](https://developer.android.com/reference/com/google/android/material/navigation/NavigationView.OnNavigationItemSelectedListener). I'd expect each individual block to return true.